### ISSUE TITLE
Only compute metric on validation set

### ISSUE THREAD

**Issue by asaluja** [allenai/allennlp#2383]
**System (please complete the following information):**
 - OS: Linux
 - Python version: 3.6.5
 - AllenNLP version: 0.7.1
 - PyTorch version: 0.4.1

**Question**
I have defined a particular metric that is expensive to compute. I don't need to compute it during training, so I'd only like to compute the metric on the validation set. Is there a way I can put this into the config file? 

Thanks,
Avneesh

**Comment by asaluja** [comment]
Thanks this answers the question. A follow-up - I wasn't aware of a class variable `self.training` as this is not easily discoverable. Where/how exactly is this set (for my own curiosity)?

**Comment by joelgrus** [comment]
this is a feature of `torch.nn.Module` (which our `Model` is a subclass of)

when you call `model.train()` it's set to true, and when you call `model.eval()` it's set to false. these both happen in the trainer.

https://pytorch.org/docs/stable/nn.html#torch.nn.Module.train
